### ISSUE TITLE
Feature Request: sticky the "Grading panel" during Manual Grading for long questions

### ISSUE THREAD

**Issue by firasm** [PrairieLearn/PrairieLearn#6778]
I'm manually grading a few multi-part "essay" type questions using PrairieLearn and obviously because the content is more than a word or a sentence, there's a bunch of scrolling that needs to be done back and forth between the content, and the grading panel. 

Feature request: Give the grading panel the `sticky-top` style so that it floats with the content. It might look a bit odd when it's overtop of the staff info box - not sure what to do about that except float around it?

Here's what it would look like at the Bottom of the page:

<img width="1361" alt="Screenshot 2022-12-18 at 3 32 53 PM" src="https://user-images.githubusercontent.com/2507459/208325397-ed60f74f-4d72-4ce9-8621-fb8c4ad68f22.png">

**Comment by firasm** [comment]
Agreed. Though, This will likely be solved by https://github.com/PrairieLearn/PrairieLearn/issues/6861 if collapsible rubric groups are introduced

**Comment by nwalters512** [comment]
Does Gradescope do independently-scrollable content + sidebar? IIRC they do, though I can't test this for myself. I'm not opposed to us doing the same, though of course I'd want to play around with it and see how it feels.

**Comment by firasm** [comment]
Yep (independently-scrollable content + sidebar), here's a quick screencast of how things look there. 

![gradescope_rubric_groups](https://github.com/user-attachments/assets/e6374f8d-7553-4876-8896-0207d6b9056a)

**Comment by jonatanschroeder** [comment]
One option is to use the offcanvas feature of Bootstrap (which is experimental in BS4 but [included in BS5](https://getbootstrap.com/docs/5.0/components/offcanvas/#backdrop)), and (unlike modals) has an option to allow the background to retain scrolling and to include/exclude a backdrop.

**Comment by nritschel** [comment]
I like Jonatan's suggestion quite a lot. I made a quick experimental edit to see how this could look like in Prairielearn and is feels pretty natural to me. 

Not sure if I should make a PR in this case since I don't think this would be used in its current form, but you can check it out if you want to get an impression

https://github.com/nritschel/PrairieLearn/tree/grading-overlay-experiment

I can also upload a screen capture of how it looks if @firasm can share his secret of how to upload them as a GIF here that is actually readable and not horribly compressed. :)

**Comment by firasm** [comment]
I use [Kap](https://getkap.co) for that !

**Comment by nritschel** [comment]
See the PR, or this GIF (thanks, @firasm!) for a demo.

![Kapture 2024-08-02 at 14 20 19](https://github.com/user-attachments/assets/d8098c4b-7ed2-4cfb-97ba-5491bf91023f)

**Comment by nwalters512** [comment]
Assuming they're small enough, you can also [upload videos](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files) (mp4 files) directly to GitHub, which are much nicer than GIFs!

**Comment by firasm** [comment]
Ooooh. 


[Aside: 
Now I'm wondering if this is a great solution for really long questions where students need to reference a bit of code in later parts. 

But this is BS5 and we're not quite there yet. Too bad. ]

**Comment by mfsilva22** [comment]
Are we quite there now? This would be a very helpful feature! (I am now super annoyed scrolling up and down as I read very long answer problems)

**Comment by nwalters512** [comment]
We are on Bootstrap 5 now, thought I don't think offcanvas is the way to go here. We used CSS Grid in the new AI question generation feature to implement side-by-side independently scrolling content, and I think that's the right thing to do here too, a la Gradescope.

**Comment by mfsilva22** [comment]
I was thinking about the AI page too! It would be great to have the same here :-)

**Comment by jonatanschroeder** [comment]
I'm bumping this issue, as I was recently working with manual grading on a question with a rubric with many items. I believe having some split panel with independent scrolling would be really useful. In particular, being able to scroll through the rubric while still seeing (1) the student answer and (2) the total points is something I really missed. Also, there needs to be some thoughts on how to access the instructor panel, which is still useful.

**Comment by ZacWarham** [comment]
Another upvote from the MDS instructors at UBC - our case is grading long multi part questions similar to Firas original suggestion

**Comment by nwalters512** [comment]
IMO the entire sidebar should be independently scrollable from the main content, and we can just leave the instructor panel (and anything else) below the rubric for now.
